src/AccordionPanel.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import cx from 'classnames';
+
+import Chunk from './Chunk';
+import Flex from './Flex';
+import FlexItem from './FlexItem';
+import Icon from './Icon';
+
+/**
+ * @module AccordionPanel
+ */
+class AccordionPanel extends React.Component {
+	/**
+	 * Document this for PR
+	 */
+	constructor(props){
+		super(props);
+		this.state = {
+			open: this.props.isOpen
+		};
+
+		this._handleToggle = this._handleToggle.bind(this);
+		this._updateHeight = this._updateHeight.bind(this);
+	}
+
+	/**
+	 * Document this for PR
+	 */
+	_handleToggle(event){
+		this.setState({
+			open: !this.state.open
+		});
+	}
+
+	_updateHeight() {
+		const content = ReactDOM.findDOMNode(this.content);
+		const animator = ReactDOM.findDOMNode(this.animator);
+		const height = this.props.isOpen ? content.clientHeight : 0;
+		const cssHeight = `${height}px`;
+
+		if (animator.style.height === cssHeight) {
+			return;
+		}
+
+		animator.style.height = cssHeight;
+	}
+
+	componentDidMount() {
+		this._updateHeight();
+	}
+
+	componentDidUpdate() {
+		this._updateHeight();
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.isOpen !== this.state.open) {
+			this.setState({ open: nextProps.isOpen });
+		}
+	}
+
+	render() {
+		const {
+			isOpen, // eslint-disable-line no-unused-vars
+			name,
+			onTriggerClick,
+			panelContent,
+			triggerIconShape,
+			triggerIconShapeActive,
+			triggerIconSize,
+			triggerIconAlign,
+			triggerLabel,
```

Comment:
@mmcgahan: that's how I originally had this, but then I can't set default props for the key values in `icon`. For example: if I want to use a default `size`, but change `shape`, values for everything but `shape` get thrown out.

If you know a way around this, let me know and I'll gladly change it